lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/cc/PrintHandler.java

```diff
@@ -454,635 +460,610 @@ public String fixupInlineReferences(String htmlString) {
-	      String filebase = "";
-	      if (filename != null) {
-		  filebase = filename;
-		  int slash = filebase.lastIndexOf("/");
-		  if (slash >= 0)
-		      filebase = filebase.substring(0, slash+1); // include trailing slash
-	      }
-
-	      String base = baseUrl;
-	      if (filename != null) {
-		  base = baseUrl + filename;
-		  int slash = base.lastIndexOf("/");
-		  if (slash >= 0)
-		      base = base.substring(0, slash+1); // include trailing slash
-	      }
-
-	      // collection id rather than URL
-	      String baseDir = baseName;
-	      if (filename != null) {
-		  baseDir = baseName + filename;
-		  int slash = baseDir.lastIndexOf("/");
-		  if (slash >= 0)
-		      baseDir = baseDir.substring(0, slash+1); // include trailing slash
-	      }
-
-	      // let importobject handle most of this, but we have to
-	      // process the attachments to make sure they're present
-
-	      Element attachmentlist = assignXml.getChild(ATTACHMENTS, assignNs);
-	      List<Element>attachments = new ArrayList<Element>();
-	      if (attachmentlist != null)
-		  attachments = attachmentlist.getChildren();
-	      List<String>attachmentHrefs = new ArrayList<String>();
-	      // note that we ignore the role attribute. No obvious way to implement it.
-	      for (Element a: attachments) {
-		  // file has to be there
-		  addFile(removeDotDot(filebase + a.getAttributeValue(HREF)));
-		  attachmentHrefs.add(a.getAttributeValue(HREF));
-	      }
-
-	      // need to prevent duplicates, as we're likely to see the same resource more than once.
-	      // Remember that we've produced this resource ID.
-	      String resourceId = resource.getAttributeValue(IDENTIFIER);
-	      String assignmentId = assignsAdded.get(resourceId);
-	      if (assignmentId == null) {
-		  AssignmentInterface a = (AssignmentInterface) assigntool;
-		  assignmentId = a.importObject(assignXml, assignNs, base, baseDir, attachmentHrefs, hide); // sakaiid for assignment
-		  if (assignmentId != null)
-		      assignsAdded.put(resourceId, assignmentId);
-	      }
-
-	      if (assignmentId!= null && !hide) {
-		  SimplePageItem item = simplePageToolDao.makeItem(page.getPageId(), seq, SimplePageItem.ASSIGNMENT, assignmentId, title);
-		  simplePageBean.saveItem(item);
-		  if (roles.size() > 0)
-		      simplePageBean.setItemGroups(item, roles.toArray(new String[0]));
-		  sequences.set(top, seq+1);
-	      }
-
-	  } else if (((type.equals(CC_WEBCONTENT) || (type.equals(UNKNOWN))) && hide) || type.equals(LAR)) {
-	      // handled elsewhere
-	  }
-	  if (type.equals(UNKNOWN)) {
-	      badTypes.add(resource.getAttributeValue(TYPE));
-	      log.debug("unknown type: " + resource.getAttributeValue(TYPE));
+	  } catch (Exception e) {
+          // when an item doesn't have the structure above it is expected to end up here, so level is debug
+		  log.debug("could not deserialize item metadata (this is expected if it is missing), {}", e.toString());
```

Comment:
It feels like this should be an `info` log unless it's a common occurrence. Not sure how anyone would be able to tell there's a problem with the structure otherwise. Did you you see this frequently? Maybe the code for retrieving the metadata tag should be separated from retrieving the structure element. This seems like it's an "all-or-nothing" retrieval. If there's no metadata then that's fine just ignore it but if there isn't a lom->general->structure maybe it's an info message?